lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -74,6 +74,15 @@ static LogicalResult customTypePrinter(Type type, AsmPrinter &os) {
                               });
         os << '>';
       })
+      .Case<EnumType>([&](auto enumType) {
+        os << "enum<";
+        llvm::interleaveComma(enumType, os, [&](EnumType::EnumElement element) {
+          os << element.name.getValue();
+          os << ": ";
+          printNestedType(element.type, os);
+        });
+        os << '>';
```

Comment:
would love it if we could ellide the `uint<0>` when possible

Reply:
Future work!